lib/Conversion/PolyToStandard/PolyToStandard.cpp

```diff
@@ -91,17 +137,36 @@ struct PolyToStandard : impl::PolyToStandardBase<PolyToStandard> {
     ConversionTarget target(*context);
     PolyToStandardTypeConverter typeConverter(context);
 
+    target.addLegalDialect<arith::ArithDialect>();
+
     // target.addIllegalDialect<PolyDialect>();
-    target.addIllegalOp<PolyFromCoeffsOp>();
-    // target.addIllegalOp<AddOp>();
+    target.addIllegalOp<PolyFromCoeffsOp, AddOp>();
     // target.addIllegalOp<MulOp>();
 
-    // TODO(https://github.com/google/heir/issues/105): add upstream type
-    // converters for func/call/return
-
     RewritePatternSet patterns(context);
-    patterns.add<ConvertPolyFromCoeffs>(typeConverter, context);
+    patterns.add<ConvertPolyFromCoeffs, ConvertAdd>(typeConverter, context);
+
+    populateFunctionOpInterfaceTypeConversionPattern<func::FuncOp>(
```

Comment:
Possible to put this in a utility function so it can be shared with the other lowerings that use the same boilerplate? Something like

```
void populateStructuralConversionPatterns(RewritePatternSet& patterns, TypeConverter& typeConverter) {
    populateFunctionOpInterfaceTypeConversionPattern<func::FuncOp>(
          patterns, typeConverter);
    target.addDynamicallyLegalOp<func::FuncOp>([&](func::FuncOp op) {
      return typeConverter.isSignatureLegal(op.getFunctionType()) &&
             typeConverter.isLegal(&op.getBody());
    });
    populateCallOpTypeConversionPattern(patterns, typeConverter);
    target.addDynamicallyLegalOp<func::CallOp>(
        [&](func::CallOp op) { return typeConverter.isLegal(op); });
    populateReturnOpTypeConversionPattern(patterns, typeConverter);
    target.addDynamicallyLegalOp<func::ReturnOp>(
        [&](func::ReturnOp op) { return typeConverter.isLegal(op); });
    populateBranchOpInterfaceTypeConversionPattern(patterns, typeConverter);
    target.markUnknownOpDynamicallyLegal([&](Operation *op) {
      return isNotBranchOpInterfaceOrReturnLikeOp(op) ||
             isLegalForBranchOpInterfaceTypeConversionPattern(op,
                                                              typeConverter) ||
             isLegalForReturnOpTypeConversionPattern(op, typeConverter);
    });
}
```

Reply:
I know BGV already uses this, right?

Reply:
Actually I already added this in lib/Conversion/Utils.cpp, so you can reuse that.